streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -476,7 +476,63 @@ public void shouldNotGetTaskWithKeyAndPartitionerWhenNotRunning() {
         globalStreams.metadataForKey("store", "key", (topic, key, value, numPartitions) -> 0);
     }
 
+    static class MyStringSerializer extends StringSerializer {
+        boolean configured = false;
+        boolean called = false;
+
+        @Override
+        public void configure(final Map<String, ?> configs, final boolean isKey) {
+            super.configure(configs, isKey);
+            configured = true;
+        }
+
+        @Override
+        public byte[] serialize(final String topic, final String data) {
+            called = true;
+            return super.serialize(topic, data);
+        }
+
+        boolean configured() {
+            return !called || configured;
+        }
+    }
+    static class MyStringDeserializer extends StringDeserializer {
+        boolean configured = false;
+        boolean called = false;
+
+        @Override
+        public void configure(final Map<String, ?> configs, final boolean isKey) {
+            super.configure(configs, isKey);
+            configured = true;
+        }
+        @Override
+        public String deserialize(final String topic, final byte[] data) {
+            called = true;
+            return super.deserialize(topic, data);
+        }
+
+        boolean configured() {
+            return !called || configured;
```

Comment:
Why we want to validate that deserialize is not called while configure is called? Ditto else where.

Reply:
The above condition should be read like this:

the serde should be configured (configured == true), otherwise the ser(de) is not used during the test.
This allows the symmetrical check shown below:
```
            if (!((MyIntSerializer) this.serializer()).configured()) return false;
            if (!((MyIntDeserializer) this.deserializer()).configured()) return false;
```

Reply:
Would it be better to extract to logic into the test but not the Serde and (De)Serilalizer, but let the `Serde` just return `serializerConfigured()` and `deserializerConfigure()` (instead of `configured()`) ?